mutation/atomic_cell.cc

```diff
@@ -79,23 +79,20 @@ compare_atomic_cell_for_merge(atomic_cell_view left, atomic_cell_view right) {
         return left.is_live() ? std::strong_ordering::less : std::strong_ordering::greater;
     }
     if (left.is_live()) {
-        auto c = compare_unsigned(left.value(), right.value()) <=> 0;
-        if (c != 0) {
-            return c;
-        }
         if (left.is_live_and_has_ttl() != right.is_live_and_has_ttl()) {
             // prefer expiring cells.
             return left.is_live_and_has_ttl() ? std::strong_ordering::greater : std::strong_ordering::less;
         }
         if (left.is_live_and_has_ttl()) {
             if (left.expiry() != right.expiry()) {
                 return left.expiry() <=> right.expiry();
-            } else {
+            } else if (right.ttl() != left.ttl()) {
                 // prefer the cell that was written later,
```

Comment:
This comment doesn't seem to reflect the need logic: the following comparison is going to order based on the TTL value - not the "expiry" (which would match the old comment).

You probably want to move this comment above line 87 and add a different comment here.

Reply:
The comment does refer to selection by ttl, although it's not obvious.
Having shorter ttl with same expiry means that the derived write time is later.
With that in mind, the cell becomes purgeable gc_grace_seconds after its write time, hence it will survive for longer (as a tombstone).